src/ucp/core/ucp_worker.c

```diff
@@ -2854,7 +2904,8 @@ ucs_status_t ucp_worker_query(ucp_worker_h worker,
     }
 
     if (attr->field_mask & UCP_WORKER_ATTR_FIELD_MAX_AM_HEADER) {
-        attr->max_am_header = ucp_am_max_header_size(worker);
+        ucp_worker_set_max_am_header(worker);
```

Comment:
no need to set here, because query is not possible when worker is not created yet

Reply:
sorry, that was added for debug purpose.